docs/getting-started-provider-dev.md

```diff
@@ -246,6 +262,113 @@ otherwise the cloud-controller-manager is not able to access k8s API.
+Here's example to explian `make upload-images`
+`upload-images` will first build images with architectures.
+For example with `REGISTRY=k8scloudprovider`, `VERSION=latest`, `IMAGE_NAMES=openstack-cloud-controller-manager`,
+and `ARCHS=arm64 amd64` will build following two images
+- `k8scloudprovider/openstack-cloud-controller-manager-arm64:latest`
+- `k8scloudprovider/openstack-cloud-controller-manager-amd64:latest`
+
+Second, it will than upload above two images.
+And finally, it will build image manifest to upload image `k8scloudprovider/openstack-cloud-controller-manager:latest` (with no architecture in image name).
+
+Image `k8scloudprovider/openstack-cloud-controller-manager:latest` will support two different `OS/ARCH` tag: `linux/arm64` and `linux/amd64`.
+
+From this point all images are ready in image repositories.
+To start using image with specific architecture. You can use `--platform` or without if you plan to use default platform:
+
+```
+docker pull --platform=linux/amd64 k8scloudprovider/openstack-cloud-controller-manager:latest
+```
+
+Or you can use image with architecture tagged in name:
+
+```
+docker pull k8scloudprovider/openstack-cloud-controller-manager-amd64:latest
+```
+
+Note that since image `k8scloudprovider/openstack-cloud-controller-manager:latest` is build by manifest,
+the image digest for image pull from `k8scloudprovider/openstack-cloud-controller-manager:latest` will be different to
+image you pull from `k8scloudprovider/openstack-cloud-controller-manager-amd64:latest`.
+
+### Only Build Image
+
+It's generally more recommended to use `make upload-images` to build and push image.
+But you might need to build image and run test on it if you're running a test job to verify that image.
+
+To prepare default environment variables:
+
+```
+# default value of VERSION is "{part of latest commit id}-dirty"
+export VERSION=latest
+# default registry is "k8scloudprovider"
+export REGISTRY=k8scloudprovider
+
+# default value for IMAGE_NAMES includes all supported images.
+export IMAGE_NAMES='openstack-cloud-controller-manager magnum-auto-healer'
+# default value for ARCHS includes all supported architectures.
+# default value for BUILD_CMDS includes all supported binaries.
+export BUILD_CMDS='openstack-cloud-controller-manager magnum-auto-healer'
+```
+
+If you not plan to build image, you can directly run `make images-all-archs` to support build for all specified images with all specified architectures.
+
+```
+export ARCHS='amd64 arm arm64 ppc64le s390x'
+make images-all-archs
+```
+
+If you plan to build image for single architecture, you can either specify `ARCHS` as `amd64` or run below commands
+
+```
+export ARCH=amd64
+make build-images
+```
+
+Or if you wish to ony build for specific architecture with specific image
+
+```
+export ARCH=amd64
+make image-openstack-cloud-controller-manager
+```
+
+Notice that if you didn't push images, there will not automatic generate image without architecture in the image name.
```

Comment:
I think its better to add like this,

`In case of building images, by default, images built are appended with this -$ARCH tag at the end. For example, for the above case,  openstack-cloud-controller-manager-amd64 is built , In case needed without it, tag it separately as required.` Followed by example

wdyt?

Reply:
done